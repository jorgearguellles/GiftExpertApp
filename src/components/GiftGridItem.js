import PropTypes from 'prop-types';

const GiftGridItem = ({ title, urlImg }) => {

  return(
    <div className="card animate__animated animate__zoomIn">
      <img alt={title} src={urlImg} />
      <p>{title}</p>
    </div>
  )
};

export { GiftGridItem };

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  urlImg: PropTypes.string.isRequired,
}

GiftGridItem.defaultProps = {
  title: "Title here",
  urlImg: "https://www.google.com",
}

