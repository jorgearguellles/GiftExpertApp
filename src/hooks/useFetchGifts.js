import { useState,useEffect } from "react"
import { getGift } from '../helpers/getGift';

const useFetchGifts = ( category ) => {
  
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGift(category)
      .then( imgs => {
        setState({
          data: imgs,
          loading: false,
        })
      })
    
  }, [ category ]);

  return state
};

export { useFetchGifts };