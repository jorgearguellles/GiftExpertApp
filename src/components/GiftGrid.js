import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftGridItem } from "./GiftGridItem";
import PropTypes from 'prop-types';

const GiftGrid = ({ category }) => {

  const {data:images, loading} = useFetchGifts(category);

  return(
    <>
      <h4>{category}</h4>

      {loading && <p>Loading...</p>}
      
      <div className="card-grid">
        {
          images.map( img => ( <GiftGridItem key={img.id} {...img} />) )
        }
      </div> 
    </>
  )
};

export { GiftGrid };

GiftGrid.propTypes = {
  category: PropTypes.string.isRequired
}