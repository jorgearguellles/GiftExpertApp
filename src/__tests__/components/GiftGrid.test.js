import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GiftGrid } from '../../components/GiftGrid';
import { useFetchGifts } from "../../hooks/useFetchGifts";
// import { GiftGridItem } from "./GiftGridItem";
jest.mock("../../hooks/useFetchGifts.js")

describe("<GiftGrid /> testing", ()=>{
  
  const category = "Matrix"

  test("1. Should to render correctly", ()=>{
    useFetchGifts.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("2. Should to show items when img loading: useFetchGifts", ()=>{
    const gifts = [{
      id: 'ABC',
      url: 'https://www.google.com',
      title: 'Title here'
    },{
      id: 'DEF',
      url: 'https://www.google.com',
      title: 'Title here'
    }];
    useFetchGifts.mockReturnValue({
      data: gifts,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category={category} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GiftGridItem').length).toBe(gifts.length)

  });

});