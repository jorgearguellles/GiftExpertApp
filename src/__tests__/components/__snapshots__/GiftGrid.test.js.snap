// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiftGrid /> testing 1. Should to render correctly 1`] = `
<Fragment>
  <h4>
    Matrix
  </h4>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GiftGrid /> testing 2. Should to show items when img loading: useFetchGifts 1`] = `
<Fragment>
  <h4>
    Matrix
  </h4>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="ABC"
      key="ABC"
      title="Title here"
      url="https://www.google.com"
      urlImg="https://www.google.com"
    />
    <GiftGridItem
      id="DEF"
      key="DEF"
      title="Title here"
      url="https://www.google.com"
      urlImg="https://www.google.com"
    />
  </div>
</Fragment>
`;
